plugins/mongodb/mongodbembed.hpp

```diff
@@ -338,54 +366,90 @@ namespace mongodbembed
+        void configure(std::unique_ptr<mongocxx::instance> && _instance) 
         {
-            _instance = std::move(instance);
-            _pool = std::move(pool);
+            instance = std::move(_instance);
         }
 
-        using connection = mongocxx::pool::entry;
+        /**
+         * @brief Creates a client object using the specified connections string. The client is
+         * added to the map of active connections.
+         * Example connectionString for different types of clusters:
+         * Standalone: mongodb+srv://<username>:<password>@cluster0.cgdbjhg.mongodb.net/?retryWrites=true&w=majority
+         * Replica Set: mongodb://<username>:<password>@mongodb0.example.com:27017,mongodb1.example.com:27017/?authSource=admin&replicaSet=myRepl
+         * Sharded: mongodb://<username>:<password>@mongos0.example.com:27017,mongos1.example.com:27017/?authSource=admin
+         * Additional arguments can be passed after the server name/s are added via ?<option0>=<setting0>&<option1>=<setting1>
+         * 
+         * @param connectionString The connection string for constructing the client object.
+         * 
+         * @param queryString A const char * holding the query string for hashing.
+         */
+        void create_connection(const char *connectionString, const char * queryString) 
+        {
+            // Use a hash of the connection string as the key to finding
+            // any connection objects
+            hash64_t key = rtlHash64VStr(queryString, rtlHash64VStr(connectionString, 0));
+
+            std::shared_ptr<mongocxx::client> client_ptr = std::make_shared<mongocxx::client>(mongocxx::client{mongocxx::uri{connectionString}}); 
+
+            {
+                CriticalBlock block(connLock);
+
+                clientConnections[key] = client_ptr;
+            }
+        }
 
         /**
-         * @brief Acquires a client from the pool. The calling thread will block until a
-         * connection is available.
+         * @brief Acquires a mongocxx client from the connections map.
+         * 
+         * @param connectionString A const char * holding the connection parameters.
          * 
-         * @return connection returns mongocxx::pool::entry
+         * @param queryString A const char * holding the query string for hashing.
+         * 
+         * @return A shared pointer to the mongocxx:client object for connecting to the database.
          */
-        connection get_connection() 
+        std::shared_ptr<mongocxx::client> get_connection(const char *connectionString, const char * queryString)
         {
-            return _pool->acquire();
+            // Get key for client object 
+            hash64_t key = rtlHash64VStr(queryString, rtlHash64VStr(connectionString, 0));
+
+            {
+                CriticalBlock block(connLock);
+
+                return clientConnections[key];
```

Comment:
note: This may not be obvious, but this will insert a null entry into the map.  In your case that is unlikely to matter, and the code is simpler as it stands.  In other cases you may not want that overhead.